lib/spack/spack/util/s3.py

```diff
@@ -27,6 +27,30 @@ def _parse_s3_endpoint_url(endpoint_url):
     return endpoint_url
 
 
+def get_mirror_s3_connection_info(connection):
+    s3_connection = {}
+
+    s3_connection_is_dict = connection and isinstance(connection, dict)
```

Comment:
Seems like `s3_connection_is_dict` would evaluate to `False` if `connection == {}`.  Is this intentional?

Reply:
Yes.  I don't see why we'd want to even attempt a check inside an empty dictionary,`{}`.  Do you see a reason to do that?

Reply:
No, I wasn't considering that much context.  I was just going by your variable name.  That's why I wondered what the intent was.

Reply:
Oh, I see.  That makes complete sense.  Do you think it's worth changing to `s3_has_connection_information` or something similar?